api/file.go

```diff
@@ -246,8 +249,21 @@ func handleImages(filenames []string, fileData [][]byte, teamId, channelId, user
 			// Create preview
 			go func() {
 				var preview image.Image
-				if width > int(utils.Cfg.FileSettings.PreviewWidth) {
-					preview = imaging.Resize(img, utils.Cfg.FileSettings.PreviewWidth, utils.Cfg.FileSettings.PreviewHeight, imaging.Lanczos)
+				ratio := float64(height) / float64(width)
```

Comment:
We use the preview image to show single image thumbnails. I think this is intentional, because preview != very big anyways. After this change has gone in, single images will not need to be clicked on. But I do agree 500x400 is a little small on a big screen.